pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -34,139 +42,101 @@
 , xkeyboard_config
 , xorg
 , zlib
+# settings and optional packages
+, cudaSupport ? false
+, cudaPackages ? null
 , lang ? "en"
+, version ? null
 }:
 
-let
-  l10n = import ./l10ns.nix {
-    inherit lib requireFile lang;
-  };
-in stdenv.mkDerivation {
-  inherit (l10n) version name src;
+let versions = import ./versions.nix { inherit lib requireFile; };
+
+    matching-versions =
+      with lib; sort compareVersions (filter
+        (v: v.lang == lang
+            && (if version == null then true else isMatching v.version version))
+        versions);
+
+    found-version =
+      if matching-versions == []
+      then throw ("No registered Mathematica version found to match"
+                  + " version=${version} and language=${lang}")
+      else lib.head matching-versions;
+
+    drv-name = ./. + "/${lib.versions.major found-version.version}.nix";
```

Comment:
```suggestion
    drv-name = ./${lib.versions.major found-version.version}.nix";
```

Reply:
This has to be written as `./.` + "/...".